src/NuGet.Clients/NuGet.VisualStudio.Common/NuGetFeedbackDiagnosticFileProvider.cs

```diff
@@ -1,24 +1,110 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
 using Microsoft.Internal.VisualStudio.Shell.Embeddable.Feedback;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Common;
 using NuGet.PackageManagement;
+using NuGet.VisualStudio.Telemetry;
+
+#nullable enable
 
 namespace NuGet.VisualStudio.Common
 {
     [Export(typeof(IFeedbackDiagnosticFileProvider))]
     public class NuGetFeedbackDiagnosticFileProvider : IFeedbackDiagnosticFileProvider
     {
+        // Don't crash VS feedback if imports fail. Maybe broken MEF is exactly what the customer is trying to report!
+        [Import(AllowDefault = true)]
+        public ISolutionManager? SolutionManager { get; set; }
+
         public IReadOnlyCollection<string> GetFiles()
         {
-            // todo: https://github.com/NuGet/Home/issues/8605
+            string filePath = GetFilePath();
+
+            // See comments on IFeedbackDiagnosticFileProvider.GetFiles
+            ThreadHelper.JoinableTaskFactory.RunAsync(() => WriteToZip(filePath))
+                .PostOnFailure(nameof(NuGetFeedbackDiagnosticFileProvider), nameof(WriteToZip));
```

Comment:
My understanding is `WriteToZip` should be the member name because it actually caused the exception. Hopefully it's clear (via stack trace) that the caller was `GetFiles`.

Reply:
Either stack trace or the person investigating the unhandled exceptions will see by reading the code.